app/scaffold-controllers/drawerController.js

```diff
@@ -32,34 +32,33 @@ function(
         return drawer;
     };
 
-    var initRightMenu = function(drawer, cartController) {
-        drawer.setRightMenu(cartController.webView);
-
-        return drawer;
-    };
-
     var initNavigationItems = function(drawer, tabItems, counterBadgeController) {
         var drawerEventHandler = function() {
             drawer.showLeftMenu();
         };
 
-        var cartEventHandler = function() {
-            drawer.showRightMenu();
-        };
+        var cartEventHandlerPromise =
+            CartModalController.init().then(function(cartModalController) {
```

Comment:
How would you feel about passing in the carteventhandler or the CartModalController into the DrawerController.init method? I feel like it would be more clear to create the cart in app.js. It seems like then we are being more specific about the app having a cart which we are only creating one of and anyone in the app can access it.